src/calendar/CalendarMonth.jsx

```diff
@@ -41,15 +41,23 @@ const CalendarMonth = React.createClass({
     let isSelectedRangeStart;
     let isSelectedRangeEnd;
 
-    if (!hideSelection && value && moment.isMoment(value) && value.isSame(d, 'day')) {
-      isSelectedDate = true;
-    } else if (!hideSelection && value && isMomentRange(value) && value.contains(d)) {
```

Comment:
The problem was that value.contains would return false when d was say midnight august 2nd and value.start was 5 pm august 2nd. This was only an issue for the start date since the end date of midnight would still be within a date range specified for a time later that same day.